### DISCUSSION TITLE
A required parameter (user) was not provided as a string in getStaticPaths

### DISCUSSION THREAD

**Discussion by helloindra** [vercel/next.js#33362] [Help]
Hello,
I am trying to create a dynamic route [user].js

its working well for ther getStaticProps, but the getStaticPath keeps on error.
Its weird, since i have many dynamic route in different folders with same structure and those all are works.

```javascript
export async function getStaticPaths() {
  let profiles = await supabase.from("profiles").select("username")

  const paths = profiles.body.map((profile) => ({
    params: { user: profile.username },
  }))

  return {
    paths,
    fallback: false, // false or 'blocking'
  }
}
```

I got this error.

**Error: A required parameter (user) was not provided as a string in getStaticPaths for /profile/[user]**

Do you guys have any ideas how to fix it ?

**Comment by tawsifhye** [comment]
params: { **user**: profile.username }, here define **user** property as per your file name

**Comment by icyJoseph** [comment]
At first glance, this looks like a type error, like profile doesn't have a username key. Did you ever verify that the profiles list you get is correct?

**Comment by JesserNawara** [comment]
try this if renaming didnt work
params: { user: profile.username.toString( ) },

**Reply by ogwok** [reply]
this worked for me thanks bro

**Reply by jplindgren** [reply]
Thanks, it worked here too.

**Reply by RicardoLelis** [reply]
the .toString() is the solution. But I want to understand why we need to cast it to str?

**Reply by mrobinson537** [reply]
Worked for me too. Thank you!
